### PR TITLE
MAPSAND-366 remove lazy initialization of HandlerThread

### PR DESCRIPTION
**Pull request by paulrenn67**
### Summary of changes

Lazy initialization of handlerThread is not really needed since anyone who creates an instance of RenderHandlerThread is going to render a map, and can cause issues with handler/handlerThread values not being aligned.

Added new tests to check start/stop works as expected.

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Comment by yunikkk** [comment]
Not sure I see how the fix prevents the ANR, could you elaborate? `isAlive` will just return `false` if `start` was not called, right?

**Comment by paulrenn67** [comment]
> Not sure I see how the fix prevents the ANR, could you elaborate? `isAlive` will just return `false` if `start` was not called, right? However I agree we might not need a separate `start` since it's used in single place and constructor call is coupled with `start`

Hi, originally handlerThread was lateinit, so calling the started method would result in a call on an uninitialized object.  This fix essentially just ensures that handlerThread is always valid.

**Comment by yunikkk** [comment]
> > ``
> 
> Hi, originally handlerThread was lateinit, so calling the started method would result in a call on an uninitialized object. This fix essentially just ensures that handlerThread is always valid.

Ok, so it would then crash, and this PR is about preventing this crash and not ANR, got it.

File: sdk/src/test/java/com/mapbox/maps/renderer/RenderHandlerThreadTest.kt [hunk 1]
```diff
@@ -39,12 +39,35 @@ class RenderHandlerThreadTest {
...
     assert(renderHandlerThread.handler == null)
   }
 
+  @Test
+  fun startStopStartTest() {
+    renderHandlerThread.start()
+    renderHandlerThread.stop()
+    try {
+      renderHandlerThread.start()
+      assert(false) { "Should have thrown IllegalThreadStateException" }
```
[the remark this answers is not in the document]
Comment (paulrenn67) [reply sdk/src/test/java/com/mapbox/maps/renderer/RenderHandlerThreadTest.kt:66]: Excellent, didn't know about `assertThrows`, that is so much cleaner!

**Comment by paulrenn67** [comment]
> > > ``
> > 
> > 
> > Hi, originally handlerThread was lateinit, so calling the started method would result in a call on an uninitialized object. This fix essentially just ensures that handlerThread is always valid.
> 
> Ok, so it would then crash, and this PR is about preventing this crash and not ANR, got it.

Yes, this should fix another crash that I noticed just with code inspection.  I think Kevin's fix is also combined here.

**Review by Chaoba — APPROVED** [review]

**Review by kiryldz — COMMENTED** [review]
@paulrenn67 🎉 on your first PR here!
Several notes:
 - please add changelog entry, smth like `Fix possible ANR when destroying renderer`
 - please sign an SLA otherwise PR could not be merged

**Review by yunikkk — APPROVED** [review]

**Comment by yunikkk** [comment]
Looks good!

**Comment by paulrenn67** [comment]
> @paulrenn67 🎉 on your first PR here! Several notes:
> 
> * please add changelog entry, smth like `Fix possible ANR when destroying renderer`
> * please sign an SLA otherwise PR could not be merged
> * rename the PR and update PR description as the scope slightly changed with your fixes

@kiryldz hopefully all ok now?

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Mapbox welcomes participation and contributions from everyone.
 
 ## Dependencies
 Bump gl-native to v10.8.0-beta.1, common to v23.0.0-beta.1. ([1568](https://github.com/mapbox/mapbox-maps-android/pull/1568))
+* Fix ANR while open and close MapView quickly with textureView set to true, fix potential crash on RenderHandlerThread creation. ([1567](https://github.com/mapbox/mapbox-maps-android/pull/1567))
```
Comment (kiryldz) [inline CHANGELOG.md:21]: this should go to `##Bug fixes` section above; 
also I'd rephrase it as this ANR not truly related to texture view - perhaps something like 
```suggestion
* Fix possible ANR when destroying renderer. ([1567](https://github.com/mapbox/mapbox-maps-android/pull/1567))
```

**Review by kiryldz — APPROVED** [review]
